File: travel_mate_backend/src/models/post.js
/**
 * 게시글 모델 (posts 테이블)
 * 커뮤니티 게시글의 제목, 본문, 이미지 URL 목록, 작성자·카테고리 참조를 저장합니다.
 */

const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');
const User = require('./user');
const PostCategory = require('./postCategory');

const Post = sequelize.define('Post', {
  id: {
    type: DataTypes.INTEGER,
    autoIncrement: true,
    primaryKey: true,
  },
  authorId: {
    type: DataTypes.STRING(255),
    allowNull: false,
    references: { model: User, key: 'email' },
    onDelete: 'CASCADE',
  },
  categoryId: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: { model: PostCategory, key: 'id' },
    onDelete: 'RESTRICT', // 카테고리에 게시글이 있으면 삭제 불가
  },
  title: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  content: {
    type: DataTypes.TEXT('long'), // LONGTEXT (CLOB, 대용량 본문)
    allowNull: false,
  },
  imageUrls: {
    type: DataTypes.JSON, // 이미지 URL 문자열 배열
    defaultValue: [],
  },
  created_at: {
    type: DataTypes.DATE,
    allowNull: false,
    defaultValue: DataTypes.NOW,
  },
  updated_at: {
    type: DataTypes.DATE,
    allowNull: false,
    defaultValue: DataTypes.NOW,
  },
}, {
  tableName: 'posts',
  timestamps: true,
  createdAt: 'created_at',
  updatedAt: 'updated_at',
});

Post.belongsTo(User, { as: 'Author', foreignKey: 'authorId' });
Post.belongsTo(PostCategory, { as: 'Category', foreignKey: 'categoryId' });
User.hasMany(Post, { foreignKey: 'authorId' });
PostCategory.hasMany(Post, { foreignKey: 'categoryId' });

module.exports = Post;

File: travel_mate_app/lib/domain/entities/post.dart
/// 커뮤니티 게시글 엔티티(제목, 내용, 이미지 URL, 작성자, 카테고리, 생성/수정일).
import 'package:equatable/equatable.dart';

class Post extends Equatable {
  final String id;
  final String authorId;
  /// 표시용 작성자 닉네임 (API에서 Author.UserProfile.nickname 등으로 내려줌).
  final String? authorNickname;
  final String title;
  final String content;
  final String category;
  final List<String> imageUrls;
  final DateTime createdAt;
  final DateTime updatedAt;

  const Post({
    required this.id,
    required this.authorId,
    this.authorNickname,
    required this.title,
    required this.content,
    required this.category,
    this.imageUrls = const [],
    required this.createdAt,
    required this.updatedAt,
  });

  @override
  List<Object?> get props => [
        id,
        authorId,
        authorNickname,
        title,
        content,
        category,
        imageUrls,
        createdAt,
        updatedAt,
      ];
}